pkgs/dart_tooling_mcp_server/test/test_harness.dart

```diff
@@ -378,3 +383,29 @@ Root rootForPath(String projectPath) =>
 final counterAppPath = p.join('test_fixtures', 'counter_app');
 
 final dartCliAppsPath = p.join('test_fixtures', 'dart_cli_app');
+
+/// A test wrapper around [LocalProcessManager] that stores commands locally
+/// instead of running them by spawning sub-processes.
+class TestProcessManager extends LocalProcessManager {
+  final commands = <String>[];
```

Comment:
Should this be a `List<String>`? It probably doesn't matter much which it is, but I would generally tend to keep it as close to the original representation as possible.

Reply:
I changed to `List<String>` so that it was easy to check equality in tests.

Reply:
Changed to `List<Object>`